src/copra/introspection.py

```diff
@@ -0,0 +1,23 @@
+from cocotb.handle import (
+    SimHandleBase, _type2cls,   # type: ignore
+)
+from cocotb import simulator
+
+def _get_array_type_info(obj: SimHandleBase, default: str) -> str:
+    """Get type info for array elements/objects."""
+    first_idx = obj.range.left if hasattr(obj, "range") else 0  # type: ignore
+    if (child_handle := getattr(obj, "_handle").get_handle_by_index(first_idx)):  # type: ignore
+        if (child_sim_type := child_handle.get_type()) in _type2cls:
+            return f"cocotb.handle.{_type2cls[child_sim_type].__name__}"
```

Comment:
Any reason to not do `type(handle[first_idx])` I'm not sure this is a great way to do it. Also I expect `get_type` to return the actual HDL type in the future.